src/frontend/ts/state/chessStore.ts

```diff
@@ -116,12 +137,25 @@ const useChessStore = create<ChessState>((set, get) => ({
       case Update.Result: {
         const resultData = data as ResultUpdate
         const gameID = resultData.gameID
-
         const displayGame = get().displayGame
-        if ((displayGame !== null) && (gameID === displayGame.info.gameID)) {
-          get().setDisplayGame(null)
+
+        const displayLastGame = async () => {
+          //
+          //  XX: make sure setting localArchive doesn't wipe an existing
+          //      archive search result.
+          //      maybe results have their own place in state?
+          //
+          //  get().setLocalArchive(await scryArchive('chess', '/archive/last/8'))
+          get().setLocalArchive(await scryArchive('chess', '/archive/all'))
+
+          const lastGame = get().localArchive.get(gameID)
+          if ((displayGame !== null) && (gameID === displayGame.info.gameID)) {
+            get().setDisplayGame(lastGame)
+          }
```

Comment:
Am I right in thinking that, upon a result, you're adding the completed game to the backend then scrying the archive to add it to `localArchive`?

I feel like this is putting too much trust in the connection between frontend and backend and it's a place where their two states could diverge. Since `activeGames` and `archivedGames` are both just `Map<gameID, info>`, it would be easy to move the game info directly from one Map to the other without it having to go through the backend.

In general the advantage of going through the backend is that you get an additional layer of verification - most of the backend is quite rigorous about checking data is correct - but I don't think that would be useful here.

Reply:
that's a good point.
changing that up will take me a little bit of time.